internal/telemetry/client.go

```diff
@@ -133,11 +133,28 @@ type Client struct {
 	newMetrics bool
 }
 
+// Logger for submission-related events
+type Logger interface {
+	Log(msg string)
+}
+
+// NewClient returns a telemetry client based on the
+// specified options
+func NewClient(opts ...Option) (client *Client) {
+	client = defaultClient()
+	for _, opt := range opts {
+		opt(client)
+	}
+	if !internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", true) {
+		client.Disabled = true
+	}
+	return client
+}
```

Comment:
Tiny nit: there should be a blank line after between this function and Client.log below.